src/lazy/encoder/binary/v1_1/value_writer.rs

```diff
@@ -967,6 +968,323 @@ impl<'value, 'top> BinaryAnnotatedValueWriter_1_1<'value, 'top> {
+    pub(crate) fn new<'a, 'b: 'a>(
+        allocator: &'b BumpAllocator,
+        buffer: &'a mut BumpVec<'b, u8>,
+        value_writer_config: ValueWriterConfig,
+        macros: &'a MacroTable,
+        parameter: Option<&'a Parameter>,
+    ) -> BinaryEExpParameterValueWriter_1_1<'a, 'b> {
+        BinaryEExpParameterValueWriter_1_1{
+            allocator,
+            buffer,
+            value_writer_config,
+            macros,
+            parameter,
+        }
+    }
+}
+
+impl<'value, 'top> ValueWriter for BinaryEExpParameterValueWriter_1_1<'value, 'top> {
+    type ListWriter = BinaryListWriter_1_1<'value, 'top>;
+    type SExpWriter = BinarySExpWriter_1_1<'value, 'top>;
+    type StructWriter = BinaryStructWriter_1_1<'value, 'top>;
+    type EExpWriter = BinaryEExpWriter_1_1<'value, 'top>;
+
+    validate_parameter_and_delegate!(
+        IonType => write_null,
+        bool => write_bool,
+        &Decimal => write_decimal,
+        &Timestamp => write_timestamp,
+        impl AsRef<str> => write_string,
+        impl AsRef<[u8]> => write_clob,
+        impl AsRef<[u8]> => write_blob,
+    );
+
+    // These types need to have ranges validated to ensure we can write the value to a tagless
+    // parameter if the parameter is tagless.
+
+    fn write_symbol(self, value: impl AsRawSymbolRef) -> IonResult<()> {
+        use crate::IonError;
+        use crate::lazy::expanded::template::ParameterEncoding;
+
+        // TODO: Support tagless types.
+        let _param = self
+            .parameter
+            .ok_or(IonError::encoding_error("unexpected parameter provided"))
+            .and_then(|p| p.expect_encoding(&ParameterEncoding::Tagged))
+            .and_then(|p| p.expect_single_expression())?;
+
+        let value_writer = BinaryValueWriter_1_1::new(
+            self.allocator,
+            self.buffer,
+            self.value_writer_config,
+            self.macros,
+        );
+        value_writer.write_symbol(value)
+    }
+
+    fn write_i64(self, value: i64) -> IonResult<()> {
+        use crate::IonError;
+        use crate::UInt;
+        use crate::lazy::expanded::template::ParameterEncoding as PE;
+
+        #[inline(never)]
+        fn error_context(name: &str, err: impl std::error::Error) -> IonError {
+            IonError::encoding_error(format!("error with value provided for '{name}': {err}"))
+        }
+
+        let param = self
+            .parameter
+            .ok_or(IonError::encoding_error("unexpected parameter provided"))
+            .and_then(|p| p.expect_single_expression())?;
+
+        let result = match param.encoding() {
+            PE::UInt8 => value
+                .try_into()
+                .and_then(|uint: UInt| FixedUInt::write_as_uint8(self.buffer, uint)),
+            PE::FlexUInt => value
+                .try_into()
+                .and_then(|uint: UInt| FlexUInt::write(self.buffer, uint))
+                .map(|_| ()),
```

Comment:
It's been a while... does the `buffer` track its own position so that it's safe to ignore the returned "number of bytes written"?

Reply:
Yep, `buffer` is a `BumpVec` which tracks writes by extending its length.